UmpleToJava/UmpleTLTemplates/JavaClassGenerator.ump

```diff
@@ -98,17 +98,27 @@ public <<# if (uClass.getIsAbstract()) { append(realSb, "{0} ", "abstract"); } #
 for (StateMachine smq : uClass.getStateMachines())
   {
     if (smq.isQueued())
-    {
-      append(realSb," implements Runnable");
+    { 
+      if (uClass.hasParentInterface() == false){
```

Comment:
@Nava2  The problem here is the implementation of code generation for queued and pooled state machines. The implemented class needs to implement Runnable. For other interfaces which are in the model, there is another method that lists all of them.